sdk/storage/azure-storage-blobs/src/blob_responses.cpp

```diff
@@ -58,4 +59,48 @@ namespace Azure { namespace Storage { namespace Blobs {
     };
   }
 
+  void ListBlobsPageResult::OnNextPage(const Azure::Core::Context& context)
+  {
+    auto contextRef = std::cref(context);
+    if (&context == &Azure::Core::Context::GetApplicationContext())
+    {
+      // Called via iterator increment, use the same context as the last call.
+      contextRef = std::cref(m_context);
+    }
```

Comment:
For v1, we may want to remain explicit and avoid this "magic" of reusing the same context.